src/runtime/auto-prompt-manager.ts

```diff
@@ -249,6 +249,10 @@ export class AutoPromptManager {
       return;
     }
 
+    if (!clientConfig.uiPredicates?.canDisplayAutoPrompt) {
```

Comment:
nit: do we want to move this to before the entitlements check to save us an entitlement check in this scenario?

Reply:
`entitlements.enablesThis()` is actually just checking the presence of entitlement, but doesn't perform any action on it. That being said, I think it makes sense to check UiPredicate first which is per-publication configuration rather than per-user data.